modules/programs/vdirsyncer-accounts.nix

```diff
@@ -0,0 +1,311 @@
+{ lib, ... }:
+
+with lib;
+
+let
+
+  collection = types.either types.str (types.listOf types.str);
+
+  remoteModule = types.submodule {
+    options = {
+      type = mkOption {
+        type = types.enum [ "caldav" "carddav" "http" "google_calendar" "google_contacts" ];
+        description = "The type of the storage.";
+      };
+
+      url = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "The url of the storage.";
+      };
+
+      timeRange = mkOption {
+        type = types.nullOr (types.submodule {
+          options = {
+            start = mkOption {
+              type = types.str;
+              description = "Start of time range to show.";
+            };
+
+            end = mkOption {
+              type = types.str;
+              description = "End of time range to show.";
+            };
+          };
+        });
+
+        default = null;
+        description = ''
+          A time range to synchronize. start and end
+          can be any Python expression that returns
+          a <literal>datetime.datetime</literal> object.
+        '';
+        example = {
+          start = "datetime.now() - timedelta(days=365)";
+          end = "datetime.now() + timedelta(days=365)";
+        };
+      };
+
+      itemTypes = mkOption {
+        type = types.nullOr (types.listOf types.str);
+        default = null;
+        description = ''
+          Kinds of items to show. The default is to
+          show everything. This depends on particular
+          features of the server, the results are not
+          validated.
+        '';
+      };
+
+      userName = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "User name for authentication.";
+      };
+
+      userNameCommand = mkOption {
+        type = types.nullOr (types.listOf types.str);
+        default = null;
+        example = [ "~/get-username.sh" ];
+        description = ''
+          A command that prints the user name to standard
+          output.
+        '';
+      };
+      password = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "Password for authentication.";
+      };
+
+      passwordCommand = mkOption {
+        type = types.nullOr (types.listOf types.str);
+        default = null;
+        example = [ "pass" "caldav" ];
+        description = ''
+          A command that prints the password to standard
+          output.
+        '';
+      };
+
+      passwordPrompt = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        example = "Password for CalDAV";
+        description = ''
+          Show a prompt for the password with the specified
+          text.
+        '';
+      };
+
+      verify = mkOption {
+        type = types.nullOr types.bool;
+        default = null;
+        description = "Verify SSL certificate. Defaults to <literal>true</literal>.";
```

Comment:
Like @teto pointed out, the "Defaults to <literal>true</literal>." is wrong and could be removed.